vault/resource_gcp_auth_backend_role.go

```diff
@@ -3,13 +3,19 @@ package vault
 import (
 	"fmt"
 	"log"
+	"regexp"
 	"strings"
 
 	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
 
 	"github.com/hashicorp/vault/api"
 )
 
+var (
+	gcpAuthBackendFromPathRegex  = regexp.MustCompile("^auth/(.+)/role/[^/]+$")
+	gcpAuthRoleNameFromPathRegex = regexp.MustCompile("^auth/.+/role/([^/]+)$")
```

Comment:
Good question, though from what I'm seeing it doesn't seem possible to specify a namespace in the backend path explicitly, since `auth/` is always prepended: 
https://github.com/hashicorp/terraform-provider-vault/blob/1db2297507bea47aee5ce6f6d729daef29de92d5/vault/resource_gcp_auth_backend_role.go#L234-L241
https://github.com/hashicorp/terraform-provider-vault/blob/1db2297507bea47aee5ce6f6d729daef29de92d5/vault/resource_gcp_auth_backend_role.go#L166-L168

Reply:
FWIW these regex's do seem to work when setting `namespace` in the provider block.

Reply:
Thanks for checking!